src/main/java/store/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -66,4 +81,21 @@ public List<BookDto> search(BookSearchParameters searchParameters) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long id) {
+        return bookRepository.findAllByCategoryId(id)
+                .stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
+    }
+
+    private Set<Category> getCategoriesFromIds(List<Long> ids) {
+        Set<Category> categories = new HashSet<>();
+
+        for (Long id : ids) {
+            categories.add(categoryMapper.toModel(categoryService.getById(id)));
```

Comment:
it is ok to use categoryRepository here, because you need category model, 
add a method to get categories by a collection of ids